widget/button.go

```diff
@@ -29,45 +29,152 @@ func (b *buttonRenderer) padding() fyne.Size {
+				case ButtonIconTrailingText:
+					// +------------------------+
+					// | Text Icon              |
+					// +------------------------+
+					labelPos.X = innerOffset.X
+					labelPos.Y = innerOffset.Y + (innerSize.Height-labelSize.Height)/2
+					iconPos.X = labelPos.X + labelSize.Width + theme.Padding()
+					iconPos.Y = innerOffset.Y + (innerSize.Height-iconSize.Height)/2
+				}
+			case ButtonAlignTrailing:
+				switch b.button.IconPlacement {
+				case ButtonIconLeadingText:
+					// +------------------------+
+					// |              Icon Text |
+					// +------------------------+
+					labelPos.X = innerOffset.X + innerSize.Width - labelSize.Width
+					labelPos.Y = innerOffset.Y + (innerSize.Height-labelSize.Height)/2
+					iconPos.X = labelPos.X - theme.Padding() - iconSize.Width
+					iconPos.Y = innerOffset.Y + (innerSize.Height-iconSize.Height)/2
+				case ButtonIconTrailingText:
+					// +------------------------+
+					// |              Text Icon |
+					// +------------------------+
+					iconPos.X = innerOffset.X + innerSize.Width - iconSize.Width
+					iconPos.Y = innerOffset.Y + (innerSize.Height-iconSize.Height)/2
+					labelPos.X = iconPos.X - theme.Padding() - labelSize.Width
+					labelPos.Y = innerOffset.Y + (innerSize.Height-labelSize.Height)/2
+				}
+			}
+			b.label.Move(labelPos)
+			b.label.Resize(labelSize)
+			b.icon.Move(iconPos)
+			b.icon.Resize(iconSize)
+		} else {
+			// Label Only
+			b.label.Move(alignedPosition(b.button.Alignment, padding, labelSize, size))
+			b.label.Resize(labelSize)
 		}
-		labelOffset := fyne.NewPos((innerSize.Width+contentWidth)/2-labelSize.Width, (innerSize.Height-labelSize.Height)/2)
-		b.label.Resize(labelSize)
-		b.label.Move(innerOffset.Add(labelOffset))
-	} else if b.icon != nil {
-		b.icon.Resize(fyne.NewSize(theme.IconInlineSize(), theme.IconInlineSize()))
-		b.icon.Move(fyne.NewPos((size.Width-theme.IconInlineSize())/2, (size.Height-theme.IconInlineSize())/2))
+	} else {
+		// Icon Only
+		b.icon.Move(alignedPosition(b.button.Alignment, padding, iconSize, size))
+		b.icon.Resize(iconSize)
+	}
+}
+
+func alignedPosition(align ButtonAlign, padding, objectSize, layoutSize fyne.Size) (pos fyne.Position) {
+	pos.Y = (layoutSize.Height - objectSize.Height) / 2
+	switch align {
+	case ButtonAlignCenter:
+		// +------------------------+
+		// |         Object         |
+		// +------------------------+
+		pos.X = (layoutSize.Width - objectSize.Width) / 2
+	case ButtonAlignLeading:
+		// +------------------------+
+		// | Object                 |
+		// +------------------------+
+		pos.X = padding.Width / 2
+	case ButtonAlignTrailing:
+		// +------------------------+
+		// |                 Object |
+		// +------------------------+
+		pos.X = layoutSize.Width - objectSize.Width - padding.Width/2
+	}
+	return
+}
+
+// applyAlignment updates the button label alignment
+func (b *buttonRenderer) applyAlignment() {
+	switch b.button.Alignment {
+	case ButtonAlignLeading:
```

Comment:
If the suggestion about a content container is picked up then this code goes away as the text is always minsize

Reply:
Removed